Zotero/Scenes/Detail/PDF/PDFCoordinator.swift

```diff
@@ -583,6 +584,18 @@ extension PDFCoordinator: PdfAnnotationsCoordinatorDelegate {
         
         self.navigationController?.present(navigationController, animated: true, completion: nil)
     }
+
+    func parseAndCacheIfNeededAttributedComment(viewModel: ViewModel<PDFReaderActionHandler>, annotation: PDFAnnotation) -> NSAttributedString? {
+        let comment = annotation.comment
+        guard !comment.isEmpty else { return nil }
+
+        if let attributedComment = viewModel.state.comments[annotation.key] {
+            return attributedComment
+        }
+
+        viewModel.process(action: .parseAndCacheComment(key: annotation.key, comment: comment))
+        return viewModel.state.comments[annotation.key]
+    }
```

Comment:
I agree this needs to be shared, but it doesn't really make sense to have this in `*Coordinator` class. I'd either move it to the view model or `PDFReaderViewController` or some other shared controller.

Reply:
Check https://github.com/zotero/zotero-ios/pull/850/commits/6efff2582f1a7c0d1f857821c57922aaffb54454